src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.WindowsSettings/Main.cs

```diff
@@ -0,0 +1,319 @@
+        /// </summary>
+        private const string _settingsFile = "WindowsSettings.json";
+
+        /// <summary>
+        /// The name of this assembly
+        /// </summary>
+        private readonly string _assemblyName;
+
+        /// <summary>
+        /// The initial context for this plugin (contains API and meta-data)
+        /// </summary>
+        private PluginInitContext? _context;
+
+        /// <summary>
+        /// The path to the icon for each result
+        /// </summary>
+        private string _defaultIconPath;
+
+        /// <summary>
+        /// Indicate that the plugin is disposed
+        /// </summary>
+        private bool _disposed;
+
+        private IEnumerable<IWindowsSetting>? _settingsList;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Main"/> class.
+        /// </summary>
+        public Main()
+        {
+            _assemblyName = Assembly.GetExecutingAssembly().GetName().Name ?? Name;
+            _defaultIconPath = _lightSymbol;
+        }
+
+        /// <summary>
+        /// Gets the localized name.
+        /// </summary>
+        public string Name => Resources.PluginTitle;
+
+        /// <summary>
+        /// Gets the localized description.
+        /// </summary>
+        public string Description => Resources.PluginDescription;
+
+        /// <summary>
+        /// Initialize the plugin with the given <see cref="PluginInitContext"/>
+        /// </summary>
+        /// <param name="context">The <see cref="PluginInitContext"/> for this plugin</param>
+        public void Init(PluginInitContext context)
+        {
+            _context = context ?? throw new ArgumentNullException(nameof(context));
+            _context.API.ThemeChanged += OnThemeChanged;
+            UpdateIconPath(_context.API.GetCurrentTheme());
+
+            var assembly = Assembly.GetExecutingAssembly();
+            var type = assembly.GetTypes().FirstOrDefault(x => x.Name == nameof(Main));
+
+            var resourceName = $"{type?.Namespace}.{_settingsFile}";
+
+            try
+            {
+                using var stream = assembly.GetManifestResourceStream(resourceName);
+                if (stream is null)
+                {
+                    throw new Exception("stream is null");
+                }
+
+                var options = new JsonSerializerOptions();
+                options.Converters.Add(new JsonStringEnumConverter());
+
+                using var reader = new StreamReader(stream);
+                var text = reader.ReadToEnd();
+
+                _settingsList = JsonSerializer.Deserialize<IEnumerable<WindowsSetting>>(text, options);
+            }
+            catch (Exception exception)
+            {
+                Log.Exception("Error loading settings JSON file", exception, typeof(Main));
+            }
+
+            if (_settingsList is null)
+            {
+                return;
+            }
```

Comment:
Let's move into helper class for better code structure: `CreateSettingsList`

Reply:
move what into own class?

Reply:
The code for reading from json. (See my general comment for the review.)

Reply:
put into separate class